### ISSUE TITLE
Inform Johnny-Five that it must not create a repl instance if deployed to flash...

### ISSUE THREAD

**Issue by rwaldron** [rwaldron/tessel-io#8]
By setting:

```js
this.repl = false;
```

In the `Tessel` constructor (before emitting "ready"), we can inform Johnny-Five that it must not create a repl instance. 

The criteria is: 

- If the cwd starts with `/app/remote-script/`

Assigning to @brianarn

**Comment by rwaldron** [comment]
Originally discovered here https://github.com/brianarn/t2-rgb-led-rainbow

**Comment by brianarn** [comment]
I took a couple of stabs at this, and am at an interesting spot:

First, I was looking at `process.cwd()` but it turns out, that doesn't return what I'd expect.

- Under `t2 run index.js`: `/root`
- Under `t2 push index.js`: `/`

Then, I switched to `__dirname` instead, which generally provides what I'd expect:

- Under `t2 run index.js`: `/tmp/remote-script`
- Under `t2 push index.js`: `/app/remote-script`

Under my current detection, under `run` the REPL will still be available (though in practice, I'm finding that it's not really usable), but under `push` it turns off, and I've verified this via my t2-rgb-led-rainbow repo as linked above.

However, I'm finding that authoring tests in this environment is difficult, as `__dirname` is per-module, so I can't easily mock it in any reasonable fashion.

- I can't just set `__dirname` since it's local to the test module
- If I set `global.__dirname` it still doesn't work, since `__dirname` appears to be a local variable

Still trying other things, but wanted to document steps thus far.
